README.md

```diff
@@ -1,5 +1,189 @@
+* `path`;
+* `options` – (optional). Possible values are:
+    * `headers`;
+    * `query`;
+    * `timeout`;
+    * `throwOnError`;`
+    * `reqContext`;
+    * `safeParseJson`;
+    * `disableKeepAlive`;`
+    * `retryConfig`, defined by:
+        * `maxAttempts`, the maximum number of times a request should be retried;
+        * `delayBetweenAttemptsInMsecs`;
+        * `statusCodesToRetry`, the status codes that trigger a retry;
+        * `retryOnTimeout`;
+    * `clientOptions`;
+    * `responseSchema`;
+    * `validateResponse`;
+    
+    The following options are applied by default:
+
+    ```ts
+    validateResponse: true,
+    throwOnError: true,
+    timeout: 30000,
+    retryConfig: {
+        maxAttemps: 1,
+        delayBetweenAttemptsInMsecs: 0,
+        statusCodesToRetry: [],
+        retryOnTimeout: false,
+    }
+    ```
+
+Additionally, `sendPost()`, `sendPut()`, `sendPutBinary()`, and `sendPatch()` also accept a `body` parameter.
+
+The response of any _send_ method will be resolved to always have `result` set, but only have `error` set in case something went wrong. See [Either](#either) for more information.
+
+## Default Logging Configuration
+
+The library provides methods to resolve the default logging configuration. Public methods available are:
+
+* `resolveLoggerConfiguration()`, which accepts as parameter an `appConfig`, defined by the `logLevel` and the `nodeEnv`. If the environment is production, the output will be logged in JSON format to be friendly with any data storage. Otherwise, the output will be logged with coloring and formatting to be visible for debugging purposes and help developers. 
+
+    The method returns a logger configuration that should be used with `pino` library as in the following example:
+
+    ```ts
+    const loggerConfig = resolveLoggerConfiguration({
+        logLevel: 'warn',
+        nodeEnv: 'production',
+    })
+
+    const logger = pino(loggerConfig)
+    ```
+
+## ConfigScope
+
+`ConfigScope` is a class that provides a way to encapsulate a single config source (e. g. `process.env`) and produce a set of values out of it, defining constraints and transformations for them.
+
+Once the class is instantiated, you can leverage the following `ConfigScope` methods:
+
+### Mandatory Configuration Parameters
+
+* `getMandatory()`, returns the value of a mandatory configuration parameter. If the value is missing, an `InternalError` is thrown. Parameters are:
+    * `param`, the configuration parameter name;
+* `getMandatoryInteger()`, returns the value of a mandatory configuration parameter and validates that it is a number. If the value is missing or is not a number, an `InternalError` is thrown. Parameters are:
+    * `param`, the configuration parameter name;
+* `getMandatoryOneOf()`, returns the value a mandatory configuration parameter and validates that it is one of the supported values. If the value is missing or is not supported, an `InternalError` is thrown. Parameters are:
```

Comment:
mention that it also serves as a type guard, narrowing down type of passed value down to one of the supported options